gunicorn/arbiter.py

```diff
@@ -508,11 +508,11 @@ def reap_workers(self):
         """\
         Reap workers to avoid zombie processes
         """
-        try:
-            while True:
-                wpid, status = os.waitpid(-1, os.WNOHANG)
+        for pid, _ in self.WORKERS.items():
```

Comment:
No need in `.items()` since you don't use dictionary value.

`for pid in tuple(self.WORKERS):` might be a better line. tuple is to force copy of pids, because `WORKERS` dict might change in the meantime.

Reply:
Used, thanks.

Reply:
The idiomatic way to iterate over the keys of a possibly-changing dict would be:

```
for pid in self.WORKERS.keys():
```

Why not use that? It's much clearer what you're trying to do.